rumqttc/src/mqttbytes/topic.rs

```diff
@@ -5,15 +5,8 @@ pub fn has_wildcards(s: &str) -> bool {
 
 /// Checks if a topic is valid
 pub fn valid_topic(topic: &str) -> bool {
-    if topic.contains('+') {
-        return false;
-    }
-
-    if topic.contains('#') {
-        return false;
-    }
-
-    true
+    // topic can't contain wildcards
+    !topic.contains('+') && !topic.contains('#')
```

Comment:
```suggestion
    if topic.contains('+') || topic.contains('#') {
        return false;
    }
   
    true
```
i feel this is more readable

Reply:
i wanted to avoid use of conditional statements as shown about and fn calls like:
```rust
!has_wildcard(topic)
```

but guess i'm just over optimizing haha, do you think fn call to `has_wildcard` be better? ( note: it might add additional fn symbols and overhead ). otherwise, I will do the suggested changes!

Reply:
update: lets avoid that fn call for now, pushed the suggested changed :+1: